component/componenttest/assert_no_error_host.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package componenttest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// assertNoErrorHost implements a component.Host that asserts that
+// there were no errors.
+type assertNoErrorHost struct {
+	t *testing.T
```

Comment:
embed nophost so you don't have to write code for nop funcs.

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector/pull/2679/commits/0c20030f853e2850994ddbc1cb06cc49dc1087f7